ambry-store/src/main/java/com/github/ambry/store/PersistentIndex.java

```diff
@@ -1537,32 +1533,21 @@ private long getAbsolutePositionInLogForOffset(Offset offset,
     LogSegment logSegment = log.getFirstSegment();
     Offset firstOffsetInJournal = journal.getFirstOffset();
     List<LogSegmentName> logSegmentNamesToReturn = new ArrayList<>();
-    if (config.storeAutoCloseLastLogSegmentEnabled) {
-      while (logSegment != null) {
-        if (firstOffsetInJournal == null) {
-          if (log.getNextSegment(logSegment) != null) {
-            logSegmentNamesToReturn.add(logSegment.getName());
-          } else {
-            break;
-          }
+    while (logSegment != null) {
+      if (firstOffsetInJournal == null) {
+        if (log.getNextSegment(logSegment) != null) {
+          logSegmentNamesToReturn.add(logSegment.getName());
         } else {
-          if (!logSegment.getName().equals(firstOffsetInJournal.getName())) {
-            logSegmentNamesToReturn.add(logSegment.getName());
-          } else {
-            break;
-          }
+          break;
         }
-        logSegment = log.getNextSegment(logSegment);
-      }
-    } else {
-      while (firstOffsetInJournal != null && logSegment != null) {
+      } else {
         if (!logSegment.getName().equals(firstOffsetInJournal.getName())) {
           logSegmentNamesToReturn.add(logSegment.getName());
         } else {
           break;
         }
-        logSegment = log.getNextSegment(logSegment);
       }
+      logSegment = log.getNextSegment(logSegment);
     }
     return logSegmentNamesToReturn.size() > 0 ? logSegmentNamesToReturn : null;
```

Comment:
this is getting a bit complicated, can we just do something like
```
ArrayList<LogSegmentName> allLogSegments = new ArrayList<>();
LogSegment logSegment = log.getFirstSegment();
while(logSegment != null) {
    allLogSegment.add(logSegment.getName());
}
if (journal.isEmpty()) {
    return allLogSegments;
} else {
    Offset firstOffsetInJournal = journal.getFirstOffset();
    int index = allLogSegments.indexOf(firstOffsetInJournal.getName());
    if (index != -1) {
        return allLogSegments.subList(0, index);
    } else {
        return allLogSegments;
    }
}
```

Reply:
Updated the logic with some minor difference.